packages/workspace-resolver/src/resolve-workspace.ts

```diff
@@ -65,7 +65,8 @@ export async function resolveWorkspace(
   const pkgPath = packageJsonPath(root);
 
   const json = await readPackageJson(isRoot, workingDirToUse, pkgPath);
-  const isModularRoot = json.modular?.type === 'root';
+  const type = json.modular?.type || undefined;
```

Comment:
Do we need to cater for converting to `undefined` any `modular.type` that has been explicitly set to `null`?

```suggestion
  const type = json.modular?.type;
```